libsolidity/analysis/TypeChecker.cpp

```diff
@@ -351,6 +351,9 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 	if (_function.overrides() && _function.isFree())
 		m_errorReporter.syntaxError(1750_error, _function.location(), "Free functions cannot override.");
 
+	if (!_function.modifiers().empty() && _function.isFree())
```

Comment:
rebase problem?

Reply:
No, this is needed, before doing modifiers on free functions resulted in a parser error, now it's possible and thus needs extra checking here.

Reply:
Oh sorry, this is new right? Then the next question: Why not?

Reply:
Ok, then let's keep this for now.

Reply:
Why not have modifiers? Well, as far as I know, modifiers only work within a contract so we'd need an instance and free functions are missing that.. at this point I tried to replicated the existing behavior (I didn't want to open too many construction sites :P )

Reply:
Yes, it's fine.